core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemListener.java

```diff
@@ -369,11 +369,17 @@ public String getIdFromRawMessage(F rawMessage, Map<String, Object> threadContex
 	@Override
 	public F changeProcessState(F message, ProcessState toState, String reason) throws ListenerException {
 		try {
-			if (!fileSystem.exists(message) || !knownProcessStates().contains(toState)) {
+			if (!getFileSystem().exists(message) || !knownProcessStates().contains(toState)) {
 				return null; // if message and/or toState does not exist, the message can/will not be moved to it, so return null.
 			}
 			if (toState==ProcessState.DONE || toState==ProcessState.ERROR) {
-				return FileSystemUtils.moveFile(fileSystem, message, getStateFolder(toState), isOverwrite(), getNumberOfBackups(), isCreateFolders());
+				return FileSystemUtils.moveFile(getFileSystem(), message, getStateFolder(toState), isOverwrite(), getNumberOfBackups(), isCreateFolders());
+			}
+			if (toState==ProcessState.INPROCESS && isFileTimeSensitive() && getFileSystem() instanceof IWritableFileSystem) {
+				F movedFile = getFileSystem().moveFile(message, getStateFolder(toState), false);
+				String newName = getFileSystem().getCanonicalName(movedFile)+"-"+(DateUtils.format(getFileSystem().getModificationTime(movedFile)).replace(":", "_"));
+				F renamedFile = getFileSystem().toFile(newName);
+				return FileSystemUtils.renameFile((IWritableFileSystem<F>) getFileSystem(), movedFile, renamedFile, overwrite, getNumberOfBackups());
```

Comment:
you cannot use overwrite or numberofbackups here, because then the new file replaces the one that was already in process.
Please use 'false' and '0'.
In case of failure, move the file to error